statsmodels/regression/tests/test_mixed_glm.py

```diff
@@ -0,0 +1,350 @@
+    library(lme4)
+    m = glmer(y ~ 0 + x1 + x2 + (1 | g), family=poisson(link=sqrt), data=df)
+    logLik(m)
+    summary(m)$coef
+    VarCorr(m)
+
+    logLik -118.5216
+    """
+
+    np.random.seed(313)
+    endog, exog, groups = gen_mixed('Poisson', 30)
+
+    sqrt_link = sm.families.links.sqrt
+    model = MixedGLM(endog, exog, groups=groups, family=sm.families.Poisson(link=sqrt_link))
+    par = MixedGLMParams.from_components(fe_params=np.asarray([0.5840612,0.5596115]),
+                                         cov_re=np.asarray([[1.2337**2]]))
+    logLik = model.loglike(par)
+    np.testing.assert_allclose(logLik, -118.5216, rtol=1e-4)
+
+def test_laplace_loglike_gamma_log():
+    """
+    Python:
+    np.random.seed(313)
+    endog, exog, groups = gen_mixed('Gamma', 30)
+    mat = np.hstack((endog[:, None], groups[:, None], exog))
+    np.savetxt('mat.txt', mat)
+
+    R:
+    df = read.table('mat.txt')
+    names(df) = c('y', 'g', 'x1', 'x2')
+    library(lme4)
+    m = glmer(y ~ 0 + x1 + x2 + (1 | g), family=Gamma(link=log), data=df)
+    logLik(m)
+    summary(m)$coef
+    VarCorr(m)
+
+    logLik -61.37757
+    """
+
+    np.random.seed(313)
+    endog, exog, groups = gen_mixed('Gamma', 30)
+
+    loglink=sm.families.links.log
+    model = MixedGLM(endog, exog, groups=groups, family=sm.families.Gamma(loglink))
+    par = MixedGLMParams.from_components(fe_params=np.asarray([0.9920529, 0.9011868]),
+                                         cov_re=np.asarray([[0.59920**2]]))
+    logLik = model.loglike(par)
+    np.testing.assert_allclose(logLik, -61.3776, rtol=1e-6)
+
+# Can't check Gaussian since it uses the concentrated log-likelihood
+
+import statsmodels.genmod.families.links as links
+
+# Family instances
+class CheckFamily(object):
+
+    family = sm.families.Family
+    test_links = []
+
+    def test_links_derivs(self):
+        for lnk in self.test_links:
+            try:
+                self.like_hess_grad(self.family, lnk)
+            except Exception as e:
+                e.args += ("Link type: " + str(lnk),)
+                raise
+
+    def like_hess_grad(self, fam, lnk):
+        """
+        Test the function, gradient and Hessian of the joint
+        log-likelihood of the random effects and the data.
+        """
+
+        fa = fam(link=lnk)
+
+        np.random.seed(423)
+        endog, exog, groups = gen_mixed(fa.__class__.__name__, 10)
+
+        model = MixedGLM(endog, exog, groups=groups, family=fa)
+
+        # Set some parameters
+        fe_params = np.random.normal(size=2)
+        cov_re = np.abs(np.random.normal(size=(1,1)))
+        scale = 1.0
+
+        self.like_hess_grad_helper(model, fe_params, cov_re, scale)
+
+    def test_links_derivs2(self):
+        for lnk in self.test_links:
+            try:
+                self.like_hess_grad2(self.family, lnk)
+            except Exception as e:
+                e.args += ("Link type: " + str(lnk),)
+                raise
```

Comment:
we get a nicer failure message if instead of using `try ... except` we just add the link to the err_msg in the asserts